[ADD] l10n_ar_edi: Electronic Invoice functional guide for Argentina (#594)
odoo/documentation-user: accounting/localizations/argentina.rst

Configure Master data
---------------------

Electronic Invoice Credentials
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Environment
***********

The AFIP infrastructure is replicated in two separate environments, Testing and Production.

Testing is provided so that the Companies can test their developments until they are ready to move 
into the Production environment. As these two environments are completely isolated from each other, 
the digital certificates of one instance are not valid in the other one. 

Go to :menuselection:`Accounting --> Settings --> Argentinian Localization` to select the environment:

.. image:: media/argentina_edi_01.png
   :align: center

AFIP Certificates
*****************
The electronic invoice and other afip services work with WebServices (WS) provided by the AFIP. 

In order to enable communication with the AFIP, the first step is to request a Digital Certificate
if you don’t have one already.   

#. Generate certificate Sign Request (Odoo).

>    When this option is selected a file with extension .csr  (certificate signing request)

Reviewer: ```suggestion
#. Generate certificate Sign Request (Odoo). When this option is selected a file with extension .csr (certificate signing request)
```
or
```suggestion
#. | Generate certificate Sign Request (Odoo).
   | When this option is selected a file with extension .csr (certificate signing request)
```
depending on the intended render.

Reviewer: Fixed

   is generated to be used the AFIP portal to request the certificate. 

>   .. image:: media/argentina_edi_02.png

Reviewer: Under-indented by 1 space since you're now using `#.` instead of `-` for the list elements. The image currently renders as a quote.

Author: Fix included in the last commit.

#. Generate Certificate (AFIP). 

>    Access the AFIP portal and follow the instructions described in the next document 

Reviewer: Same as above about line breaks.

Author: Fix included in the last commit.

Author: Fix included in the last commit.

   in order to get a certificate. `Get AFIP Certificate
   <http://www.afip.gob.ar/ws/WSAA/wsaa_obtener_certificado_produccion.pdf>`_.
   
#. Upload Certificate and Private Key (Odoo).

>    Once the certificate has been generated, it needs to be uploaded in Odoo, using the 

Reviewer: Same as above about line breaks.

Author: Fix included in the last commit.

   pencil next in the field “Certificado” and selecting the corresponding file. 

>   .. image:: media/argentina_edi_03.png

Reviewer: Same as above about image indentation.

Author: Fix included in the last commit.

.. image:: media/argentina15.png
   :align: center

Special Use Cases
~~~~~~~~~~~~~~~~~

Invoices for Services
^^^^^^^^^^^^^^^^^^^^^

> or electronic invoices that include Services, the AFIP requires to report the  service starting

Author: It was "For..." it will be corrected in the PR.

and ending date, this information can be filled in the tab “Other Info”: 

.. image:: media/argentina_edi_06.png
   :align: center

If  the dates are not selected manually before the invoice is validated, the values will be
filled automatically considering the beginning and day of the invoice month:

.. image:: media/argentina_edi_07.png
   :align: center

Exportation Invoices
^^^^^^^^^^^^^^^^^^^^
The invoices related to Exportation transactions required a Journal that used the AFIP POS 
System “Expo Voucher - Web Service” so the proper document type be associated:

.. image:: media/argentina_edi_08.png
   :align: center

When the customer selected in the Invoice has set the AFIP responsibility type as “Cliente / 
Proveedor del Exterior” or “IVA Liberado – Ley Nº 19.640”, Odoo automatically assigned:

- Journal related to the exportation Web Service.
- Exportation document type .
- Fiscal position: Compras/Ventas al exterior.
- Concepto AFIP:  Products / Definitive export of goods.
- Exempt Taxes. 

.. image:: media/argentina_edi_09.png
   :align: center

.. note::

>    The Exportation Documents required the Incoterm in  “Other Info” -> Accounting section:

Reviewer: ```suggestion
   The Exportation Documents required the Incoterm in :menuselection:`Other Info --> Accounting`:
```

   
.. image:: media/argentina_edi_13.png
   :align: center
   
   
Fiscal Bond
^^^^^^^^^^^
The Electronic Fiscal bond is used for those who invoice capital goods and wish to access 
the benefit of the Electronic Tax Bonds granted by the Ministry of Economy.

For these transactions it’s important to have into consideration the next requirements:

- Currency (according to parameter table) and invoice quotation.
- Taxes.
- Zone.
- Detail each item.

  - Code according to the Common Nomenclator of Mercosur (NCM).
  - Complete description.
  - Unit Net Price.
  - Quantity.
  - Unit of measurement.
  - Bonus.
  - VAT rate. 

Electronic Credit Invoice MiPyme (FCE) 
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> ``Invoices`` 

Reviewer: \`\`This syntax\`\` is generally used to highlight technical terms. For important terms, we use \*\*bold\*\* as you did [here](https://www.odoo.com/documentation/user/13.0/accounting/localizations/argentina.html#introduction).

Don't forget about the non-effective line break too !

There are several document types classified as Mipyme also known as Electronic Credit 
Invoice (FCE in spanish), which is used to impulse the SME, its purpose is to develop 
a mechanism that improves the financing conditions of these companies and allows them 
to increase their productivity, through the early collection of credits and receivables 
issued to their clients and / or vendors. 

For these transactions it’s important to have into consideration the next requirements:

- Specific document types (201, 202, 206, etc).
- The emisor should be eligible by the AFIP to MiPyme transactions. 
- The amount should be bigger than 100,000 ARS. 
- A bank account type CBU must be related to the emisor, otherwise the invoice can’t 
  be validated, having these errors messages for example:  
  
.. image:: media/argentina_edi_10.png
   :align: center

``Credit& Debit Notes``

> When creating a Credit/Debit note related to a FCE document, it is important take the

Reviewer: Same as above about highlighting and line break.

next points into consideration:

- Use the Credit and Debit Note buttons, so the correct reference of the originator 
  document passed to the note. 

.. image:: media/argentina_edi_11.png
   :align: center
   
- The document letter should be the same than the originator document (either A or B).
- The same currency as the source document must be used.  When using a secondary currency 
  there is an exchange difference  if the currency rate is different between the emission day 
  and the payment date, it’s possible to create a credit/debit note to decrease/increase the 
  amount to pay in ARS. 

In the workflow we can have two scenarios:

1. The FCE is rejected so the Credit Note should have the field “FCE, is Cancellation?” as True. 

> 2. The Credit Note, is created with the negative amount to annulate the FCE document, 

Reviewer: ```suggestion
#. The FCE is rejected so the Credit Note should have the field “FCE, is Cancellation?” as True. 
#. The Credit Note, is created with the negative amount to annulate the FCE document, 
```
Prefer automatic number for easier editing later on.

.. image:: media/argentina17.png
   :align: center

The vendor bill number is structured in the same way that the invoices with the difference 
that the document sequence is input by the user: “Document Prefix - Letter -  Document number".

Validate Vendor Bill number in AFIP
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
As most companies have internal controls to verify that the vendor bill is related to an AFIP
valid document,  an automatic validation can be set in Accounting -> Settings -> 

> Argentina localization -> “Validate document in the AFIP”, considering the following levels: 

Reviewer: menuselection

- ``Not available:`` The verification is not done (this is the default value).
- ``Available:``  The verification  is done, in case the number is not valid 
  it only raises a warning but it allows you to post the vendor bill. 
- ``Required:`` The verification  is done and it doesn't allow the user to 

>   post the vendor bill if the document number is not valid.

Reviewer: Same as above about highlighting.

.. image:: media/argentina_edi_16.png
   :align: center

> **How to use it in Odoo:**

Reviewer: Having a line in bold and starting with "How to" is generally the sign that the line is either not required or should be a heading.

> This tool incorporates in the vendor bill a new "Verify on AFIP" button located 

Reviewer: Same as above about line break.

next to the AFIP Authorization code. 

.. image:: media/argentina_edi_17.png
   :align: center

In case it’s not a valid AFIP authorization the value “Rejected” will be 
displayed and the details of the validation will be added to the chatter.

.. image:: media/argentina_edi_18.png
   :align: center

Special Use cases
~~~~~~~~~~~~~~~~~
Untaxed Concepts
^^^^^^^^^^^^^^^^
There are some transactions that include items that are not part of the VAT base amount, 
this is commonly used in fuel and gasoline invoices. 

**How to register them in Odoo:**

> The vendor bill will be registered using 1 item for each product that is part of the VAT

Reviewer: Same as above about heading and line break.

base amount and an additional item to register the amount of  the Exempt concept:

.. image:: media/argentina_edi_19.png
   :align: center

Perception Taxes
^^^^^^^^^^^^^^^^
**How to register them in Odoo:**

> The vendor bill will be registered using 1 item for each product that is part of the 

Reviewer: Same as above about heading and line break.

.. image:: media/argentina17.png
   :align: center

The vendor bill number is structured in the same way that the invoices with the difference 
that the document sequence is input by the user: “Document Prefix - Letter -  Document number".

Validate Vendor Bill number in AFIP
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
As most companies have internal controls to verify that the vendor bill is related to an AFIP
valid document,  an automatic validation can be set in :menuselection:`Accounting --> Settings --> 
Argentinian Localization --> Validate document in the AFIP`, considering the following levels: 

- **Not available:** The verification is not done (this is the default value).
- **Available:**  The verification  is done, in case the number is not valid 
  it only raises a warning but it allows you to post the vendor bill. 
- **Required:** The verification  is done and it doesn't allow the user to 

>   post the vendor bill if the document number is not valid.

Reviewer: ```suggestion
- **Not available**: The verification is not done (this is the default value).
- **Available**: The verification  is done, in case the number is not valid 
  it only raises a warning but it allows you to post the vendor bill. 
- **Required**: The verification  is done and it doesn't allow the user to 
  post the vendor bill if the document number is not valid.
```
(for in GitHub commit)
